micrometer-core/src/main/java/io/micrometer/core/instrument/logging/LoggingMeterRegistry.java

```diff
@@ -152,13 +152,28 @@ protected void publish() {
                                     loggingSink.accept(print.id() + " throughput=" + print.rate(count) +
                                             " mean=" + print.time(timer.mean(getBaseTimeUnit())));
                                 },
-                                meter -> loggingSink.accept(print.id() + StreamSupport.stream(meter.measure().spliterator(), false)
-                                        .map(ms -> ms.getStatistic().getTagValueRepresentation() + "=" + decimalOrNan(ms.getValue())))
+                                meter -> loggingSink.accept(writeMeter(meter, print))
                         );
                     });
         }
     }
 
+    private String writeMeter(Meter meter, Printer print) {
+        return print.id() + StreamSupport.stream(meter.measure().spliterator(), false)
+                .map(ms -> {
+                    String msLine = ms.getStatistic().getTagValueRepresentation() + "=";
+                    switch (ms.getStatistic()) {
+                        case TOTAL:
+                        case MAX:
+                        case VALUE:
+                            return msLine + print.value(ms.getValue());
+                        default:
+                            return msLine + decimalOrNan(ms.getValue());
+                    }
+                })
+                .collect(joining(" ", " ", ""));
```

Comment:
Perhaps a comma and space (`, `) would more clearly delineate measurements. What do you think?
Could you also share a whole line output for a `Meter` after this change? Or adding a test would be even better but not required.

Reply:
```suggestion
                .collect(joining(", ", " ", ""));
```

Reply:
Sure!